modules/dind/dind.go

```diff
@@ -0,0 +1,118 @@
+package dind
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/mount"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+var (
+	// containerPorts {
+	defaultDockerDaemonPort = "2375/tcp"
+)
+
+// DinDContainer represents the Docker in Docker container type used in the module
+type DinDContainer struct {
+	testcontainers.Container
+}
+
+// Deprecated: use Run instead
+// RunContainer creates an instance of the K3s container type
+func RunContainer(ctx context.Context, opts ...testcontainers.ContainerCustomizer) (*DinDContainer, error) {
+	return Run(ctx, "docker:28.0.1-dind", opts...)
+}
+
+// Run creates an instance of the K3s container type
+func Run(ctx context.Context, img string, opts ...testcontainers.ContainerCustomizer) (*DinDContainer, error) {
+	req := testcontainers.ContainerRequest{
+		Image: img,
+		ExposedPorts: []string{
+			defaultDockerDaemonPort,
+		},
+		Privileged: true,
+		HostConfigModifier: func(hc *container.HostConfig) {
+			hc.CgroupnsMode = "host"
+			hc.Tmpfs = map[string]string{
+				"/run":     "",
+				"/var/run": "",
+			}
+			hc.Mounts = []mount.Mount{}
+		},
+		Cmd: []string{
+			"dockerd", "-H", "tcp://0.0.0.0:2375", "--tls=false",
+		},
+		Env: map[string]string{
+			"DOCKER_HOST": "tcp://localhost:2375",
+		},
+		WaitingFor: wait.ForLog("API listen on [::]:2375"),
```

Comment:
suggestion: let's use a strategy that checks for the listening port instead, which could be more robust that the log entry